jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -76,17 +80,65 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
             raise ValueError(f"Length of data labels list ({len(data_label)}) is different"
                              f" than length of list of data ({len(data)})")
 
-    with app.data_collection.delay_link_manager_update():
         # these are used to build a combined spectrum with all
         # input spectra included (taken from https://github.com/spacetelescope/
         # dat_pyinthesky/blob/main/jdat_notebooks/MRS_Mstar_analysis/
         # JWST_Mstar_dataAnalysis_analysis.ipynb)
-        wlallorig = []
-        fnuallorig = []
-        dfnuallorig = []
+        wlallorig = []  # to collect wavelengths
+        fnuallorig = []  # fluxes
+        dfnuallorig = []  # and uncertanties (if present)
+
+        for spec in data:
+            for wlind in range(len(spec.spectral_axis)):
+                wlallorig.append(spec.spectral_axis[wlind].value)
+                fnuallorig.append(spec.flux[wlind].value)
+
+                # because some spec in the list might have uncertainties and
+                # others may not, if uncert is None, set to list of NaN. later,
+                # if the concatenated list of uncertanties is all nan (meaning
+                # they were all nan to begin with, or all None), it will be set
+                # to None on the final Spectrum1D
+                if spec.uncertainty[wlind] is not None:
+                    dfnuallorig.append(spec.uncertainty[wlind].array)
+                else:
+                    dfnuallorig.append(np.nan)
+
+    # if the entire uncert. array is Nan and the data is not, model fitting won't
+    # work (more generally, if uncert[i] is nan/inf and flux[i] is not, fitting will
+    # fail, but just deal with the all nan case here since it is straightforward).
+    # set uncerts. to None if they are all nan/inf, and display a warning message.
+    set_nans_to_none = False
+    if isinstance(data, SpectrumList):
+        uncerts = dfnuallorig  # alias these for clarity later on
+        if uncerts is not None:
+            if not np.any(uncerts):
+                uncerts = None
+                set_nans_to_none = True
+    else:
+        if len(data) > 1:
+            # otherwise, data should be a list containing a single [Spectrum1D]
+            # check just to be safe.
+            m = 'Only supported for one data entry, if multiple spectra use SpectrumList.'
+            raise NotImplementedError(m)
+
+        if data[0].uncertainty is not None:
+            uncerts_finite = np.isfinite(data[0].uncertainty.array)
+            if not np.any(uncerts_finite):
+                data[0].uncertainty = None
+                set_nans_to_none = True
+
+    if set_nans_to_none:
+        # alert user that we have changed their all-nan uncertainty array to None
+        msg = 'All uncertainties are nonfinite, replacing with uncertainty=None.'
+        info_msg = SnackbarMessage(msg, color="warning", sender=app)
+        app.hub.broadcast(info_msg)
```

Comment:
I have been shortening old calls like this one to one line. No reason to assign 2 variables to this one snackbar call, as they will never be used again.
```suggestion
        msg = 'All uncertainties are nonfinite, replacing with uncertainty=None.'
        info_msg = SnackbarMessage(msg, color="warning", sender=app)
        app.hub.broadcast(info_msg)
```